### PR TITLE
Initial role commit

### PR DESCRIPTION
**Pull request by danpawlik**
This commit provides Ansible role, that deploys Microshift [1] service on RHEL base host.
The role has been ported from RDO project [2].

[1] https://microshift.io/
[2] https://review.rdoproject.org/r/plugins/gitiles/rdo-infra/rdo-infra-playbooks/+/refs/heads/master/roles/rdo-infra/microshift/

### CODE REVIEW

File: tasks/firewall.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  package_facts:
+
+- name: Add exception to the firewall
+  when: "'firewalld' in ansible_facts.packages"
+  block:
+    - name: Set ports
+      become: true
+      ansible.posix.firewalld:
+        port: "{{ zj_rule_details }}"
+        permanent: true
```
Comment (cjeanner) [inline tasks/firewall.yaml:12]: I'd avoid `permanent: true` here - we probably don't want it to persist reboot? Or a parameter maybe?

File: tasks/firewall.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        state: enabled
+      loop:
+        - 80/tcp
+        - 443/tcp
+        - 5353/udp
+      loop_control:
+        loop_var: zj_rule_details
+
+    - name: Add exception to source
+      become: true
```
Comment (cjeanner) [inline tasks/firewall.yaml:22]: ditto

File: tasks/firewall.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      ansible.posix.firewalld:
+        source: 10.42.0.0/16
+        zone: trusted
+        permanent: true
```
Comment (cjeanner) [inline tasks/firewall.yaml:26]: ditto

File: tasks/pre.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      br_netfilter
+    dest: /etc/modules-load.d/k8s.conf
+
+- name: Enable module
+  become: true
+  ansible.builtin.shell: |
+    modprobe br_netfilter
+    modprobe overlay
+
+- name: Update sysctl
```
Comment (cjeanner) [inline tasks/pre.yaml:40]: Maybe use https://docs.ansible.com/ansible/latest/collections/ansible/posix/sysctl_module.html - it has a `reload` parameter among interesting features.

File: tasks/pv.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+- name: Get worker nodes
+  ansible.builtin.shell: |
+    oc get node -o name -l node-role.kubernetes.io/worker | sed -e 's|node/||' | head -c-1 | tr '\n' ','
+  tags:
+    - skip_ansible_lint
+  register: _worker_nodes
+
+- name: Add PV
+  ansible.builtin.shell: |
```
Comment (cjeanner) [inline tasks/pv.yaml:57]: can't we get it as a template?

**Comment by cjeanner** [comment]
> given this is the inital import i would generally prefer to merge this as is and then refine in follow up PRs
> 
> one thing you should add is an apache2 licence file to the root of the repo to match the license we use for all out repos we maintain in this github org. we are trying to keep them the same unless it's a fork of a different operator.

the license is already in actually - I selected it upon repository creation.

**Comment by cjeanner** [comment]
@SeanMooney those are mostly tiny bits - if we can get them in during the import, that's all we won't have to check back later :).

Guess we'll also introduce some testing, just to be on the safe side - that can come a bit later. Maybe the ci-framework may help on that regard.

File: tasks/pv.yaml (hunk 5 above)
Comment (danpawlik) [reply tasks/pv.yaml]: done

File: tasks/pre.yaml (hunk 4 above)
Comment (danpawlik) [reply tasks/pre.yaml:37]: done

File: tasks/firewall.yaml (hunk 3 above)
Comment (danpawlik) [reply tasks/firewall.yaml]: done

File: tasks/firewall.yaml (hunk 1 above)
Comment (danpawlik) [reply tasks/firewall.yaml]: parametrized is fine.
Done

File: tasks/pre.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  ansible.builtin.lineinfile:
+    dest: /etc/hosts
+    regexp: ".*{{ fqdn }}$"
+    line: "{{ ansible_default_ipv4.address }} {{ fqdn }}"
+    state: present
+
+# From Kubernetes setup - https://kubernetes.io/docs/setup/production-environment/container-runtimes/
+- name: Add modules to be loaded
+  become: true
+  ansible.builtin.copy:
```
[the remark this answers is not in the document]
Comment (danpawlik) [reply tasks/pre.yaml]: done

**Review by raukadah — APPROVED** [review]
Almost all the suggestions are already addressed. SInce it is the initial import, let's merge it

**Review by arxcruz — APPROVED** [review]

**Pull request merged**